### PR TITLE
Built out ChartSelector component

### PR DESCRIPTION
**Pull request by akstarre**
Built out ChartSelector component to change receive and pass chart selections to state. added minimal styling for clarity.

### CODE REVIEW

File: src/components/Chart/index.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    },
+    y: {
+      display: false, // Removes the numbers for the y-axis
+    },
+  },
+}
+
+interface ChartDataInterface {
+
+}
```
Comment (yoloOnTheBattlefield) [inline src/components/Chart/index.tsx:35]: this is not used

File: src/components/Chart/index.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+type ChartProps = {
+  chartType: string;
+  chartData: ChartDataInterface;
+  currentChart: string;
+
+}
+
+export const Chart =({chartType, chartData, currentChart}: ChartProps) =>{
+
+
```
Comment (yoloOnTheBattlefield) [inline src/components/Chart/index.tsx]: too many new lines

File: src/app/page.tsx [hunk 3]
```diff
@@ -1,36 +1,41 @@
...
   useEffect(() => {
     dispatch(fetchCoins(currentCurrency));
   }, [currentCurrency, dispatch])
 
+  const handleChartSelection = (selection:string) => {
+    dispatch(changeChart(selection))
+  }
 
   return (
     <div className="">
```
Comment (yoloOnTheBattlefield) [inline src/app/page.tsx:30]: no empty className

File: src/app/page.tsx [hunk 4]
```diff
@@ -1,36 +1,41 @@
...
 
 
 const Home = () => {
 
   const dispatch = useDispatch<AppDispatch>();
-  const {currency: currentCurrency} = useAppSelector((state) => state.currency)
-  const {coins, loading, error} = useSelector((state: RootState) => state.marketTable);
- 
+  const {currency: currentCurrency, currentChart } = useAppSelector((state) => state.currency)
+  const {coins, loading, error} = useSelector((state: RootState) => state.marketTable);
```
Comment (yoloOnTheBattlefield) [inline src/app/page.tsx:19]: needs space around {}

File: src/app/page.tsx [hunk 5]
```diff
@@ -1,36 +1,41 @@
 "use client"
 
 import { Navbar } from "@/components/Navbar";
-import { LineChart } from "@/components/LineChart"
+import { Chart } from "@/components/Chart"
 import  MarketTable  from "../components/MarketTable/index"
+import {ChartSelector} from "../components/ChartSelector/index"
 import React, { useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
```
Comment (yoloOnTheBattlefield) [inline src/app/page.tsx:8]: absolute imports come first

File: src/app/GlobalRedux/Features/CurrencySlice/currencySlice.ts [hunk 6]
```diff
@@ -16,9 +17,12 @@ const currencySlice = createSlice({
     reducers: {
             changeCurrency (state, action) {
                 state.currency = action.payload
+            },
+            changeChart (state, action) {
+                state.currentChart = action.payload
             }
```
Comment (yoloOnTheBattlefield) [inline src/app/GlobalRedux/Features/CurrencySlice/currencySlice.ts:23]: indetation is too wide

File: src/app/GlobalRedux/Features/CurrencySlice/currencySlice.ts [hunk 7]
```diff
@@ -16,9 +17,12 @@ const currencySlice = createSlice({
...
     }
 })
 
-export const {changeCurrency} = currencySlice.actions
+export const {changeCurrency, changeChart} = currencySlice.actions
```
Comment (yoloOnTheBattlefield) [inline src/app/GlobalRedux/Features/CurrencySlice/currencySlice.ts:27]: needs space around {}

File: src/app/page.tsx [hunk 8]
```diff
@@ -1,36 +1,41 @@
...
 import { fetchCoins } from "@/app/GlobalRedux/Features/MarketTable/marketTableSlice"
+import { changeChart } from "@/app/GlobalRedux/Features/CurrencySlice/currencySlice";
 import { AppDispatch, RootState } from '@/app/GlobalRedux/store';
 import  {useAppSelector} from "./GlobalRedux/store"
```
Comment (yoloOnTheBattlefield) [inline src/app/page.tsx:12]: needs space around {}

File: src/app/page.tsx [hunk 9]
```diff
@@ -1,36 +1,41 @@
...
       <Navbar />
       <div className="flex w-full justify-around items-center p-4">
+        <div><ChartSelector coins={coins} currentChart={currentChart} currentCurrency={currentCurrency} handleChartSelection={handleChartSelection}/></div>
         <div className="relative w-[calc(50%-16px)]">
-          <LineChart />
+          {/* <Chart chartType="line" currentChart={currentChart}/> */}
         </div>
         <div className="relative w-[calc(50%-16px)]">
-          <LineChart />
+          {/* <Chart chartType="bar" currentChart={currentChart}/> */}
```
Comment (yoloOnTheBattlefield) [inline src/app/page.tsx:38]: no commented out code without a reason

File: src/app/page.tsx [hunk 10]
```diff
@@ -1,36 +1,41 @@
...
+  }
 
   return (
     <div className="">
       <Navbar />
       <div className="flex w-full justify-around items-center p-4">
+        <div><ChartSelector coins={coins} currentChart={currentChart} currentCurrency={currentCurrency} handleChartSelection={handleChartSelection}/></div>
         <div className="relative w-[calc(50%-16px)]">
-          <LineChart />
+          {/* <Chart chartType="line" currentChart={currentChart}/> */}
```
Comment (yoloOnTheBattlefield) [inline src/app/page.tsx:35]: no commented out code without a reason

File: src/components/Chart/index.tsx [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+type ChartProps = {
+  chartType: string;
+  chartData: ChartDataInterface;
+  currentChart: string;
+
+}
+
+export const Chart =({chartType, chartData, currentChart}: ChartProps) =>{
```
Comment (yoloOnTheBattlefield) [inline src/components/Chart/index.tsx:44]: needs space around {}

File: src/components/ChartSelector/index.tsx [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+        <div 
+          ref={scrollContainerRef} 
+          className="flex whitespace-nowrap overflow-x-auto scrollbar-hide" 
+          style={{ maxWidth: '100vw' }}
+        >
+          {coins.map((coin, index) => {
+            const isCurrent = coin.name === currentChart;
+            return (
+              <div
+                key={index}
```
Comment (yoloOnTheBattlefield) [inline src/components/ChartSelector/index.tsx]: don;t use index as key

File: src/components/ChartSelector/index.tsx [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+        });
+      }
+    };
+  
+    return (
+      <div className="relative">
+        <div 
+          ref={scrollContainerRef} 
+          className="flex whitespace-nowrap overflow-x-auto scrollbar-hide" 
+          style={{ maxWidth: '100vw' }}
```
Comment (yoloOnTheBattlefield) [inline src/components/ChartSelector/index.tsx]: no inline styles

File: src/components/ChartSelector/index.tsx [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+          {coins.map((coin, index) => {
+            const isCurrent = coin.name === currentChart;
+            return (
+              <div
+                key={index}
+                className={`${
+                  isCurrent ? 'bg-blue-300' : 'bg-blue-800'
+                } p-4 rounded-lg shadow-lg cursor-pointer m-2 inline-block`}
+                onClick={() => handleSelection(coin.name)}
+                style={{ minWidth: '200px' }}
```
Comment (yoloOnTheBattlefield) [inline src/components/ChartSelector/index.tsx]: no inlone styles

File: src/components/ChartSelector/index.tsx [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+          className="absolute top-1/2 right-0 bg-blue-300 text-white p-4 rounded-full"
+          onClick={scrollRight}
+          style={{ zIndex: 1 }}
+        >
+          ➡
+        </button>
+        <button 
+          className="absolute top-1/2 left-0 bg-blue-300 text-white p-4 rounded-full"
+          onClick={scrollLeft}
+          style={{ zIndex: 1 }}
```
Comment (yoloOnTheBattlefield) [inline src/components/ChartSelector/index.tsx]: no inline styles

File: src/components/ChartSelector/index.tsx [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+              </div>
+            );
+          })}
+        </div>
+        <button 
+          className="absolute top-1/2 right-0 bg-blue-300 text-white p-4 rounded-full"
+          onClick={scrollRight}
+          style={{ zIndex: 1 }}
+        >
+          ➡
```
Comment (yoloOnTheBattlefield) [inline src/components/ChartSelector/index.tsx]: use an svg

File: src/components/ChartSelector/index.tsx [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+                <div>{coin.current_price} {currentCurrency}</div>
+                <div>{coin.price_change_percentage_24h_in_currency}%</div>
+              </div>
+            );
+          })}
+        </div>
+        <button 
+          className="absolute top-1/2 right-0 bg-blue-300 text-white p-4 rounded-full"
+          onClick={scrollRight}
+          style={{ zIndex: 1 }}
```
Comment (yoloOnTheBattlefield) [inline src/components/ChartSelector/index.tsx]: no inline styles

File: src/components/MarketListItem/index.tsx [hunk 18]
```diff
@@ -3,8 +3,32 @@
...
+  index: number; 
+
+}
+
+type DropdownProps = {
+  handleSelection: (currency: string) => void;
+  currentCurrency: string;
+}
+
+
```
Comment (yoloOnTheBattlefield) [inline src/components/MarketListItem/index.tsx:30]: too many new lines

File: src/components/MarketListItem/index.tsx [hunk 19]
```diff
@@ -3,8 +3,32 @@
...
+  market_cap: number;
+  circulating_supply: number;
+  total_supply: number;
+}
 
-export const MarketListItem = ({ coin, index }) => {
+type MarketListItemProps = {
+  coin: Coin;
+  index: number; 
+
```
Comment (yoloOnTheBattlefield) [inline src/components/MarketListItem/index.tsx:22]: new line not needed

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/components/Chart/index.tsx (hunk 1 above)
Comment (akstarre) [reply src/components/Chart/index.tsx:35]: It's prepped bc idk how my data is going to look just yet. I haven't passed it to the chart component. I just prepped it.

**Pull request merged**